src/main/java/org/jboss/remoting3/remote/IntIndexHashMap.java

```diff
@@ -200,14 +200,18 @@ public boolean add(final V v) {
     }
 
     public <T> T[] toArray(final T[] a) {
-        final ArrayList<T> list = new ArrayList<T>(size());
-        list.addAll((Collection<T>) this);
+        final ArrayList<V> list = new ArrayList<V>(size());
```

Comment:
Surely this didn't make a difference?  In fact this is probably better written as: ``return new ArrayList<T>((Collection<T>) this).toArray(a);``